### ISSUE TITLE
Misleading code in tutorial

### ISSUE THREAD

**Issue by benob** [pfnet/chainer#777]
At the end of the tutorial, the following code assumes that 10000 is a multiple of batchsize. If it is not the case, the last 
batch will be smaller, but the accuracy and loss are computed with the full batch size. This results in the accuracy being overestimated. It should be fixed by using len(x.data) instead of batchsize.

```
>>> sum_loss, sum_accuracy = 0, 0
>>> for i in range(0, 10000, batchsize):
...     x = Variable(x_test[i : i + batchsize])
...     t = Variable(y_test[i : i + batchsize])
...     loss = model(x, t)
...     sum_loss += loss.data * batchsize
...     sum_accuracy = model.accuracy.data * batchsize
...
>>> mean_loss = sum_loss / 10000
>>> mean_accuracy = sum_accuracy / 10000
```

**Comment by wkentaro** [comment]
fixed now? https://github.com/pfnet/chainer/blob/master/examples/mnist/train_mnist.py#L97-L98

**Comment by benob** [comment]
Yes, thanks
